File: backend/services/secret_manager.py
import os
from google.cloud import secretmanager

project_id = os.getenv("GOOGLE_CLOUD_PROJECT", 'interrogation-room')

def getsecret(secret_name: str) -> str:
    """指定されたシークレット名のシークレットをGoogle Cloud Secret Managerから取得する"""
    client = secretmanager.SecretManagerServiceClient()
    secret_name = f"projects/{project_id}/secrets/{secret_name}/versions/latest"
    response = client.access_secret_version(request={"name": secret_name})
    if not response.payload.data:
        raise ValueError(f"Secret '{secret_name}' not found in environment variables.")
    return response.payload.data.decode("UTF-8")

File: app/src/lib/secret-manager.ts
import { SecretManagerServiceClient } from '@google-cloud/secret-manager';

const projectId = process.env.GOOGLE_CLOUD_PROJECT || 'interrogation-room';

export async function getSecret(secretName: string): Promise<string> {
  /**
   * 指定されたシークレット名のシークレットをGoogle Cloud Secret Managerから取得する
   */
  const client = new SecretManagerServiceClient();
  const name = `projects/${projectId}/secrets/${secretName}/versions/latest`;
  
  try {
    const [version] = await client.accessSecretVersion({ name });
    const payload = version.payload?.data;
    if (!payload) {
      throw new Error(`Secret '${secretName}' not found`);
    }
    return payload.toString();
  } catch (error) {
    console.error(`Error accessing secret ${secretName}:`, error);
    throw new Error(`Failed to access secret: ${secretName}`);
  }
}
